rsqueakvm/plugins/immutability/pointers.py

```diff
@@ -0,0 +1,133 @@
+"""
+Immutable W_PointersObject Implementation.
+
+.. data:: POINTERS_CLASSES
+A list of all immutable W_PointersObject subclasses. The position of each class
+in the list correlates to its number of storage slots (`0` no storage, `1` has
+one storage slot, ...). The last class in the list is an immutable
+W_PointersObject subclass with variable storage size.
+
+.. data:: POINTERS_CLASS_ITER
+Unrolling iterable of `POINTERS_CLASSES`.
+"""
+
+from rsqueakvm.model.base import W_AbstractObjectWithIdentityHash
+from rsqueakvm.model.pointers import W_PointersObject
+from rsqueakvm.plugins.immutability import immutable_class
+from rsqueakvm.storage_classes import ClassShadow
+
+from rpython.rlib import rerased
+from rpython.rlib.unroll import unrolling_iterable
+
+STORAGE_ATTR_TEMPLATE = "storage_%d"
+
+
+@immutable_class
+class W_AbstractImmutable_PointersObject(W_PointersObject):
+    """Abstract `W_PointersObject` subclass for immutable pointers objects."""
+    _attrs_ = []
+    _immutable_fields_ = []
+    repr_classname = ('%s_AbstractImmutable' %
+                      W_PointersObject.repr_classname)
+
+    def __init__(self, space, w_cls):
+        """
+        Initialize immutable pointers object, but avoid initializing storage
+        and reuse `self.strategy` slot to store class shadow.
+        """
+        W_AbstractObjectWithIdentityHash.__init__(self)
```

Comment:
Well, it's there, it's just not mentioned that this is why we don't use the immediate super constructor...